pkg/apis/etcd/v1alpha1/types.go

```diff
@@ -40,6 +46,13 @@ type EtcdStorage struct {
 
 // EtdcStorageSpec is the spec for a EtcdStorage resource
 type EtdcStorageSpec struct {
+	// CACertConfigMap contains name and namespace of ConfigMap where CA serving certificate for etcdproxy pod
+	// is supposed to be deployed. Usually it is in aggregated API server namespace.
+	CACertConfigMap []CertificateSource `json:"caCertConfigMap"`
+
+	// ClientCertSecret contains name and namespace of Secret where client certificate and key for etcdproxy pod
+	// is supposed to be deployed. Usually it is in aggregated API server namespace.
+	ClientCertSecret []CertificateSource `json:"clientCertSecret"`
```

Comment:
I recommend not re-using the type.  We've been having people switch over to specific reference types in the API.

Reply:
I have decided to go with `CABundleDestination` and `ClientCertificateDestination`. Destination suffix makes more sense for me, compared to Source. If you have other name ideas, let me know.